tests/test_public_interface.py

```diff
@@ -564,3 +564,100 @@ def test_humanize_bytes_formats_nicely():
     assert mack.humanize_bytes(1234567890) == "1.23 GB"
     assert mack.humanize_bytes(1234567890000) == "1.23 TB"
     assert mack.humanize_bytes(1234567890000000) == "1.23 PB"
+
+
+def test_find_composite_key(tmp_path):
+    path = f"{tmp_path}/find_composite_key"
+    data = [
+        (
+            'c054f1c7-3765-49d6-aa76-debd6e76691c', 'users_bronze_dlt', 'correct_schema', 60000, 0, 'COMPLETED', 0,
+            1000000,
+            '2021-10-06T14:07:00.000+0000'),
+        ('d5d76478-ff24-4bca-aede-c69f31b5b35e', 'user_silver_dlt', 'valid_id', 50000, 400, 'COMPLETED', 0, 1000000,
+         '2021-10-06T14:07:00.000+0000'),
+        (
+            '4b07c459-f414-492a-9f80-640a741c12c6', 'user_gold_dlt', 'valid_income', 60000, 1600, 'COMPLETED', 1600,
+            100000,
+            '2021-10-07T14:02:00.000+0000'),
+        ('c054f1c7-3765-49d6-aa76-debd6e76691c', 'spend_silver_dlt', 'valid_id', 70000, 500, 'COMPLETED', 0, 1000000,
+         '2021-10-08T14:09:00.000+0000'),
+        ('c054f1c7-3765-49d6-aa76-debd6e76691c', 'users_bronze_dlt', 'correct_schema', 70000, 1000, 'COMPLETED', 0,
+         1000000, '2021-10-08T14:09:00.000+0000'),
+        ('d5d76478-ff24-4bca-aede-c69f31b5b35e', 'user_silver_dlt', 'valid_id', 60000, 1400, 'COMPLETED', 0, 1000000,
+         '2021-10-08T14:09:00.000+0000'),
+        ('4b07c459-f414-492a-9f80-640a741c12c6', 'user_gold_dlt', 'valid_age', 60000, 1600, 'COMPLETED', 1600, 100000,
+         '2021-10-08T14:09:00.000+0000'),
+        ('4b07c459-f414-492a-9f80-640a741c12c6', 'user_gold_dlt', 'valid_score', 60000, 1600, 'COMPLETED', 1600, 100000,
+         '2021-10-08T14:09:00.000+0000'),
+    ]
+    df = spark.createDataFrame(data, ['id', 'dataset', 'name', 'passed_records', 'failed_records', 'status_update',
+                                      'dropped_records', 'output_records', 'timestamp'
+                                      ])
+    df.write.format("delta").save(path)
+
+    delta_table = DeltaTable.forPath(spark, path)
+
+    composite_keys = mack.find_composite_key(delta_table.toDF(), ['passed_records', 'failed_records', 'status_update',
+                                                                  'dropped_records', 'output_records'])
+
+    expected_keys = ['id', 'name', 'timestamp']
+
+    assert composite_keys == expected_keys
+
```

Comment:
Let's add another example to show the behavior when a composite key candidate cannot be identified.

Reply:
Added another example to show the behavior when composite keys could not be identified.